packages/client/auth-core/src/contexts/authlogin.tsx

```diff
@@ -78,18 +77,23 @@ export function AuthProvider({ children, crossmint, setJwtToken, appearance }: A
         }
     }, [crossmint.jwt]);
 
+    const value = useMemo(() => ({ login, logout, jwt: crossmint.jwt }), [login, logout]);
+
     return (
-        <AuthContext.Provider value={{ login, logout, jwt: crossmint.jwt }}>
+        <AuthContext.Provider value={value}>
             {children}
-            {modalOpen && (
-                <AuthModal
-                    baseUrl={crossmintService.crossmintBaseUrl}
-                    setModalOpen={setModalOpen}
-                    setJwtToken={setJwtToken}
-                    apiKey={crossmint.apiKey}
-                    appearance={appearance}
-                />
-            )}
+            {modalOpen
+                ? createPortal(
```

Comment:
mind dropping some context on what a portal is and why we need it?

Reply:
done lmk if it covers enough